### DISCUSSION TITLE
Create item next to dropdown selection flow

### DISCUSSION THREAD

**Discussion by tomtobac** [twilio-labs/paste#1100] [Q&A] [answered]
Hey team 👋 , I am working on refactor from a page on console which the user needs to select a public key from a dropdown and has the option to create a new one from the plus button which it will trigger a modal with a simple form

Do you have any suggested for a pattern like that?

This is the legacy pattern:
![image](https://user-images.githubusercontent.com/10562610/105225924-984aed00-5b5f-11eb-8754-35e20b3e320b.png)

This is how I try to do it with paste, not very prod 🤣 
![image](https://user-images.githubusercontent.com/10562610/105226141-e233d300-5b5f-11eb-86eb-de40cf964dc1.png)

**Comment by corinne-nagel** [comment]
Hi @tomtobac , it looks like what you're working on is fairly specific to Console, and while we can give guidance for component usage, in terms of hierarchy/layout/product flow I'm not sure we have enough insight to make a recommendation. If possible, please sign up for one of our [office hours](https://docs.google.com/spreadsheets/d/10vD91zdwr9hjgDOV3HSGnoC0LgvYw-thmn3X_kvW0xk/edit#gid=800758837). Some things I'm seeing from a higher level component perspective:

- It looks like you're using the legacy Console styles, so please update those to match the newer Paste theme seen on the [Paste site](https://paste.twilio.design/).
- Ideally use only [one primary action per page](https://paste.twilio.design/components/button) (so that add button could be secondary if that isn't the most important action on the page)

- We do have patterns that you can reference such as Create:
https://paste.twilio.design/patterns/create

- For guidance on the command bar please see this post: 
https://github.com/twilio-labs/paste/discussions/1094 

If you can, please test with your users/reach out to a Console designer and do what is best for your product experience 🙂

**Comment by SiTaggart** [comment, marked as answer]
Have you considered just adding a "Create new encryption credentials" option into the combobox as an option to select?

It's a pretty common pattern, and removes the need for a button entirely. Upon selecting the "Create new ${x}" you launch the create modal.

**Reply by tomtobac** [reply]
Yes, totaly, thats was the advice of our designer. Thanks and sorry for the delay!
